Sources/AsyncHTTPClient/ConnectionPool.swift

```diff
@@ -224,6 +224,34 @@ final class ConnectionPool {
         fileprivate var closePromise: EventLoopPromise<Void>
 
         var closeFuture: EventLoopFuture<Void>
+
+        func removeIdleConnectionHandlersForLease() -> EventLoopFuture<Connection> {
+            return self.channel.eventLoop.flatSubmit {
+                self.removeHandler(IdleStateHandler.self).flatMap { () -> EventLoopFuture<Bool> in
+                    self.channel.pipeline.handler(type: IdlePoolConnectionHandler.self).flatMap { idleHandler in
+                        self.channel.pipeline.removeHandler(idleHandler).flatMapError { _ in
+                            self.channel.eventLoop.makeSucceededFuture(())
+                        }.map {
+                            idleHandler.timeoutClosed.load()
+                        }
+                    }.flatMapError { error in
+                        if let channelError = error as? ChannelPipelineError, channelError == .notFound {
```

Comment:
when does this happen? Don't we always add that handler? If there's a legit case where this would happen, please add a comment when that is

Reply:
Fixed